linkerd/opencensus/src/lib.rs

```diff
@@ -56,87 +61,111 @@ where
         }
     }
 
-    async fn run(mut self) {
-        'reconnect: loop {
-            let (request_tx, request_rx) = mpsc::channel(1);
-            let mut svc = TraceServiceClient::new(self.client.clone());
-            let req = grpc::Request::new(request_rx);
+    async fn run(self) {
+        let Self {
+            client,
+            node,
+            mut spans,
+            mut metrics,
+        } = self;
+
+        // Holds the batch of pending spans. Cleared as the spans are flushed.
+        // Contains no more than MAX_BATCH_SIZE spans.
+        let mut accum = Vec::new();
```

Comment:
nit/TIOLI: We could just preallocate to `MAX_BATCH_SIZE` here so the vec never gets resized?